tests/tacacs/utils.py

```diff
@@ -297,10 +296,16 @@ def check_server_received(ptfhost, data):
             E501 : Line too long. Following sed command difficult to split to multiple line.
     """
     sed_command = "sed -n 's/.*-> 0x\(..\).*/\\1/p'  /var/log/tac_plus.log | sed ':a; N; $!ba; s/\\n//g' | grep '{0}'".format(hex_string)   # noqa W605 E501
-    res = ptfhost.shell(sed_command)
-    logger.info(sed_command)
-    logger.info(res["stdout_lines"])
-    pytest_assert(len(res["stdout_lines"]) > 0)
+
+    # After tacplus service receive data, it need take some time to update to log file.
+    def log_exist(ptfhost, sed_command):
+        res = ptfhost.shell(sed_command)
+        logger.info(sed_command)
+        logger.info(res["stdout_lines"])
+        return len(res["stdout_lines"]) > 0
+
+    exist = wait_until(timeout, 1, 0, log_exist, ptfhost, sed_command)
```

Comment:
>sed_command [](http://example.com/codeflow?start=58&length=11)

In extreme case, will current testcase `sed_command` catch the previous testcase's log line?

Reply:
This will not happen because UT already truncate log files before test. and TACACS logs are stored in 3 log file: syslog, /var/log/tac_plus.log and /var/log/tac_plus.acct